src/string.cr

```diff
@@ -3350,23 +3355,45 @@ class String
     nil
   end
 
+  private def index_short(hash_type, offset : Int32, pointer : UInt8*, end_pointer : UInt8*, search, &)
+    search_hash = hash_type.new(0)
+    hash = hash_type.new(0)
+    mask = hash_type.new(0)
+
+    search.each_byte do |b|
+      search_hash = (search_hash << 8) | b
+      hash = (hash << 8) | pointer.value
+      mask = (mask << 8) | 0xff
+      pointer += 1
+    end
+
+    while true
```

Comment:
IMO `loop do` is more crystalistic, whatever this mean.
```suggestion
    loop do
```

Reply:
it isn't